oauth2-authorization-server/src/test/java/org/springframework/security/oauth2/server/authorization/oidc/authentication/OidcClientRegistrationAuthenticationProviderTests.java

```diff
@@ -627,6 +650,60 @@ public void authenticateWhenValidAccessTokenThenReturnClientRegistration() {
 		assertThat(clientRegistrationResult.getRegistrationAccessToken()).isEqualTo(jwt.getTokenValue());
 	}
 
+	@Test
+	public void authenticateWhenSettingPasswordEncoder() {
```

Comment:
Please remove this test and add an integration test to `OidcClientRegistrationTests`.

Here's a template of the test:

```java
// gh-1056
@Test
public void requestWhenClientRegistersWithSecretThenClientAuthenticationSuccess() throws Exception {
	// TODO Update the PasswordEncoder @Bean to PasswordEncoderFactories.createDelegatingPasswordEncoder()
	this.spring.register(AuthorizationServerConfiguration.class).autowire();

	// TODO
	// Create new client with client_credentials grant
	OidcClientRegistration clientRegistration = OidcClientRegistration.builder() ...

	// Register client
	OidcClientRegistration clientRegistrationResponse = registerClient(clientRegistration);

	// TODO
	// Use the newly registered client to obtain an access token using the client_credentials grant
	// if the access token is obtained then client authentication succeeded

}

```